Develop

Dolnys opened the pull request:
[DEMO LINK](https://Dolnys.github.io/layout_catalog/)
[TEST REPORT LINK](https://Dolnys.github.io/layout_catalog/report/html_report/)

anpiet commented:
- there should always be 4 cards in a row (not 2, 3 or 5). When width goes below 1024px, there is less cards in rows.

anpiet reviewed (changes requested):
there should always be 4 cards in a row (not 2, 3 or 5). When width goes below 1024px, there is less cards in rows.

Dolnys commented:
> there should always be 4 cards in a row (not 2, 3 or 5). When width goes below 1024px, there is less cards in rows.

 Good point, I added the min width of the content to 1024 px so now there should always be 4 elements in a row.

mvjl000 commented on src/index.html:
```diff
@@ -1,12 +1,291 @@
+        class="card__button-link"
+        data-qa="card-hover">Buy
+        </a>
+      </div>
+      <div class="card">
```
Checklist point 1

I would add empty line between cards, your html would look way cleaner ;)

mvjl000 commented on src/styles/utils/_variables.scss:
```diff
@@ -0,0 +1,7 @@
+$color-main-accent: #060b35;
+$color-secondary: #616070;
+$color-accent-blue: #00acdc;
+$color-border-card: #f3f3f3;
+$img-product-height: 134px;
+$img-product-width: 160px;
+$img-star-height-width: 16px;
```
Checklist point 11

Create variables only for values that are being used more than once

mvjl000 commented on src/styles/blocks/content_container.scss:
```diff
@@ -0,0 +1,9 @@
+.main {
+  display: flex;
+  min-width: 1024px;
+  max-width: 1200px;
+  justify-content: center;
+  flex-wrap: wrap;
+  padding: 50px 40px;
+  margin: 0 auto;
+}
```
It is said in the readme that if you are familiar with grid it is mandatory to use it.
I'm aware of that you have done this task a while ago but I would like you to refactor your layout to grid. And additionally add media queries for [RWD](https://developer.mozilla.org/en-US/docs/Learn/CSS/CSS_layout/Responsive_Design) - remember about mobile first approach.
Here are the [grid layout instrutions](https://github.com/mate-academy/layout_catalog/blob/master/grid.md) from the readme

mvjl000 commented on src/styles/blocks/navigation.scss:
```diff
@@ -0,0 +1,63 @@
+    font-weight: 500;
+    display: flex;
+    align-items: center;
+    transition: all 0.3s;
```
List all the properties that you want to have transition.

mvjl000 replied:
@Dolnys one thing I forgot about - add smooth transitions and animations. Check the readme

choeqq commented on src/styles/blocks/card.scss:
```diff
@@ -0,0 +1,108 @@
+  border-radius: 5px;
+  background-color: #ffff;
+  transition: transform 0.3s;
+  // margin-right: 48px;
+  // margin-bottom: 46px;
```
please remove comments before pushing commit

Dolnys replied:
Done

choeqq commented on src/styles/blocks/content_container.scss:
I am thinking of changing file name to `main` 🤔 - typically we use same file name as BEM block - I think it would be better to keep it like in other files, beacuse technically it is `content_container` but we've agreed upon file structure convention - so I wouldn't really change that 😉

Dolnys replied:
I changed the 'main' class to 'card', 'card' to 'card__content', and removed whole content__container file

choeqq reviewed (changes requested):
Code looks really good, but some tests are still failling due to size of cards differences, please try to fix it and you're good to go 🚀

Dolnys commented:
I have deleted the comments and changed the file name, but I don't know how to get rid of these errors. I have spent several hours on it, and it still doesn't pass the tests. After consulting with Anastasia, I should leave the code as it is.
